timdex_dataset_api/metadata.py

```diff
@@ -249,18 +264,30 @@ def setup_duckdb_context(self) -> DuckDBPyConnection:
             3. Create additional metadata views as needed.
 
         The resulting, in-memory DuckDB connection is used for all metadata queries.
+
+        If a static database file is not found, a configured DuckDB connection is still
+        returned.
         """
+        start_time = time.perf_counter()
+
         conn = duckdb.connect()
         self.configure_duckdb_connection(conn)
 
-        if self.database_exists():
-            self._attach_database_file(conn)
-        else:
+        if not self.database_exists():
             logger.warning(
                 f"Static metadata database not found @ '{self.metadata_database_path}'. "
                 "Please recreate via TIMDEXDatasetMetadata.recreate_database_file()."
             )
+            return conn
+
+        self._attach_database_file(conn)
+        self._create_append_deltas_view(conn)
+        self._create_records_union_view(conn)
+        self._create_current_records_view(conn)
```

Comment:
This is the primary work of setting up our DuckDB context.  Each of these builds tables and views in the in-memory connection.  Note that each one of these is virtually instant, where no data is transferred; they are all effectively "lazy" tables and views.